### PR TITLE
Add versioning using sbt-git and sbt-release

### PR DESCRIPTION
**Pull request by oliverlockwood**
Addresses #3.

I've added a tag named `kafka-configurator-0.1.0` to the `HEAD` of `master`.

With this tag in place (i.e. the last release can be inferred as version 0.1.0), the following SBT behaviour happens:
>kafka-configurator(versioning)> version
[info] 0.2.0-SNAPSHOT
kafka-configurator(versioning)> showReleaseVersion
0.2.0
[success] Total time: 0 s, completed 13-Jun-2017 15:59:10
kafka-configurator(versioning)> showNextVersion
0.3.0-SNAPSHOT
[success] Total time: 0 s, completed 13-Jun-2017 15:59:12
kafka-configurator(versioning)>

which looks all as it should to me, according to Maven convention.

I have manually tested the `sbt ';release with-defaults'` task, and this does what it should - runs the tests, tags the repo with the release version, and pushes the tags.

Actually doing release logic (building and publishing an artifact) is out of scope for this PR.

### CODE REVIEW

File: build.sbt [hunk 1]
```diff
@@ -19,14 +22,23 @@ val dependencies = Seq(
...
+    gitSettings,
+    releaseSettings
+  )
+
+// TODO: Move these into Release.scala
+// Useful tasks to show what versions would be used if a release was done.
+val showReleaseVersion = taskKey[String]("the future version once releaseNextVersion has been applied to it")
+val showNextVersion = taskKey[String]("the future version once releaseNextVersion has been applied to it")
+showReleaseVersion := { val rV = releaseVersion.value.apply(version.value); println(rV); rV }
+showNextVersion := { val nV = releaseNextVersion.value.apply(version.value); println(nV); nV }
```
Comment (oliverlockwood) [inline build.sbt]: If someone can help me move these 4 lines I'd be grateful.  I ran out of time.

File: build.sbt [hunk 2]
```diff
@@ -19,14 +22,23 @@ val dependencies = Seq(
 
 val root = (project in file("."))
   .enablePlugins(JavaAppPackaging)
+  .enablePlugins(BuildInfoPlugin, GitVersioning, GitBranchPrompt)
   .settings(
     organization := "com.sky",
     scalaVersion := "2.12.1",
-    version := "0.1.0-SNAPSHOT",
```
Comment (oliverlockwood) [inline build.sbt]: damn this line!

File: project/Aliases.scala [hunk 3]
```diff
@@ -0,0 +1,9 @@
+import sbt._
+
+object Aliases {
+
+  lazy val defineCommandAliases = {
+    addCommandAlias("ciBuild", ";clean; test") ++
+      addCommandAlias("ciRelease", ";clean; release with-defaults")
```
Comment (oliverlockwood) [inline project/Aliases.scala]: I think Travis could use `ciBuild` for normal builds.  We can run `ciRelease` somehow (TBD) to do a release build.

File: project/Git.scala [hunk 4]
```diff
@@ -0,0 +1,30 @@
...
+  }
+
+  lazy val gitSettings = Seq(
+    showCurrentGitBranch,
+    git.baseVersion := "0.0.0",
+    git.useGitDescribe := true,
+    git.gitTagToVersionNumber := {
+      case VersionRegex(v, "") => Some(v)
+      case VersionRegex(v, "SNAPSHOT") => Some(s"${bumpVersion(v)}-SNAPSHOT")
+      case VersionRegex(v, s) => Some(s"${bumpVersion(v)}-SNAPSHOT")
```
Comment (lacarvalho91) [inline project/Git.scala]: this can be `(v, _)` as we don't use `s`

File: build.sbt [hunk 5]
```diff
@@ -19,14 +23,24 @@ val dependencies = Seq(
 
 val root = (project in file("."))
   .enablePlugins(JavaAppPackaging)
+  .enablePlugins(BuildInfoPlugin, GitVersioning, GitBranchPrompt)
```
Comment (lacarvalho91) [inline build.sbt]: shall we just add these to the line above?

File: project/Release.scala [hunk 6]
```diff
@@ -0,0 +1,45 @@
...
+  // so it doesn't write to a version.sbt file.
+  lazy val setReleaseVersion: ReleaseStep = setVersionOnly(_._1)
+
+  def setVersionOnly(selectVersion: Versions => String): ReleaseStep = { st: State =>
+    val vs = st.get(ReleaseKeys.versions).getOrElse(sys.error("No versions are set! Was this release part executed before inquireVersions?"))
+    val selected = selectVersion(vs)
+
+    st.log.info("Setting version to '%s'." format selected)
+    val useGlobal = Project.extract(st).get(releaseUseGlobalVersion)
+    val versionStr = (if (useGlobal) globalVersionString else versionString) format selected
```
Comment (lacarvalho91) [inline project/Release.scala]: don't think this is used

File: project/Git.scala (hunk 4 above)
Comment (paoloambrosio-skyuk) [inline project/Git.scala]: Agree: both `VersionRegex(v, "SNAPSHOT")` and `VersionRegex(v, s)` do the same thing and would match `VersionRegex(v, _)`

File: build.sbt (hunk 1 above)
Comment (lacarvalho91) [inline build.sbt]: should be able to remove .apply so it would look like `releaseVersion.value(version.value)` not a big deal tho

Comment (oliverlockwood) [inline build.sbt]: Thanks to @paoloambrosio-skyuk for his time here.

**Pull request merged**